public/apps/account/log-out-button.tsx

```diff
@@ -17,28 +17,39 @@ import React from 'react';
 import { EuiButtonEmpty } from '@elastic/eui';
 import { HttpStart } from 'opensearch-dashboards/public';
 import { logout, samlLogout } from './utils';
+import { AuthType, OPENID_AUTH_LOGOUT } from '../../../common';
 
 export function LogoutButton(props: {
-  authType: string;
+  authType: string[];
   http: HttpStart;
   divider: JSX.Element;
   logoutUrl?: string;
 }) {
-  if (props.authType === 'openid') {
+  const currentAuthType = sessionStorage.getItem('current_auth_type');
+
+  if (
+    currentAuthType === AuthType.OPEN_ID ||
+    (props.authType.length === 1 && props.authType[0] === AuthType.OPEN_ID)
+  ) {
+    sessionStorage.removeItem('current_auth_type');
```

Comment:
Why removing `current_auth_type`?

Reply:
current_auth_type is saved to SeesionStorage when user click OIDC or SAML login button. This value will be maintained in the SeesionStorage even through user logout from OSD. So remove this attribute from SeesionStorage when user logout with current auth type to avoid data discrepancy.

Reply:
current_auth_type will only be set when multiple authentication mode using OIDC/SAML auth handler. If user click Basic Login button, current_auth_type is undefined. browser will be redirected to /auth/logout endpoint.